src/Returns/Domain/Items/ReturnItem.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelCom\Wms\Returns\Domain\Items;
+
+use MyParcelCom\Wms\Returns\Domain\Payment\Currency;
+
+/**
+ * ReturnItem extends the ArrayObject class with getters to make
+ *  it convenient to extract data from a return coming in from MyParcel.
+ */
+readonly class ReturnItem
+{
+    public function __construct(
+        public string $externalReference,
+        public string $sku,
+        public string $name,
+        public int $quantity,
+        public int $priceAmount,
+        public Currency $currency,
+        public string $description,
+        public ?int $weight = null,
+        public ?WeightUnit $weightUnit = null,
+        public ?string $comment = null,
+        public ?string $returnReason = null,
+    ) {
+    }
+
+    public static function fromSnakeCaseArray(array $requestArray): self
```

Comment:
Same here - the case is not relevant, like I suggested in the other comment we could just name this `from` or `create`